pkg/controller/config/config_controller.go

```diff
@@ -261,169 +247,150 @@ func (r *ReconcileConfig) Reconcile(ctx context.Context, request reconcile.Reque
 		r.tracker.DisableStats()
 	}
 
+	if err := r.waca.HandleGVKsToSync(ctx, gvks, instance.Spec.Match, r.tracker, r.reader); err != nil {
+		return reconcile.Result{}, fmt.Errorf("error handling syncOnly update: %w", err)
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func containsString(s string, items []string) bool {
+	for _, item := range items {
+		if item == s {
+			return true
+		}
+	}
+	return false
+}
+
+func removeString(s string, items []string) []string {
+	var rval []string
+	for _, item := range items {
+		if item != s {
+			rval = append(rval, item)
+		}
+	}
+	return rval
+}
+
+func hasFinalizer(instance *configv1alpha1.Config) bool {
+	return containsString(finalizerName, instance.GetFinalizers())
+}
+
+func removeFinalizer(instance *configv1alpha1.Config) {
+	instance.SetFinalizers(removeString(finalizerName, instance.GetFinalizers()))
+}
+
+type WatchAwareCacheAccuator struct {
+	Registrar      *watch.Registrar
+	WatchedSet     *watch.Set
+	needsReplaySet *watch.Set
+	needsWipe      bool
+
+	CacheManager *cm.CacheManager
+}
+
+func (w *WatchAwareCacheAccuator) HandleGVKsToSync(ctx context.Context, gvks []schema.GroupVersionKind, matchers []configv1alpha1.MatchEntry, tracker *readiness.Tracker, reader client.Reader) error {
```

Comment:
All of the functionality within HandleGVKsToSync is meant to be encapsulated within the CacheManager.